### ISSUE TITLE
how to infer

### ISSUE THREAD

**Issue by blacksino** [Leminhbinh0209/CVPR24-FAS#2]
I have tried run this model on my custom dataset and results seems to be not good or did I miss some preprocessing part?

**Comment by Leminhbinh0209** [comment]
Hi,

I suppose you're inferring the model. Make sure that you load the pretrained weights with `strict=True`, and apply transformation to the cropped faces (see SAFAS for croppibg and load it by Image.open(face_path).convert('RGB')) :
```
test_transform = transforms.Compose([
            transforms.Resize((256,256)),
            transforms.ToTensor(),
            transforms.Normalize(mean=[0.485, 0.456, 0.406],
                                         std=[0.229, 0.224, 0.225])
        ])
```
probability of `live` class is `prob= model(input)` (BCE loss)

**Comment by blacksino** [comment]
yes, I did exactly what you have mentiond above.however,the result is relatively strange

```
model = get_model(config)
model = model.cuda()
ckpt = torch.load("/mnt/a/joe/code/CVPR24-FAS/CVPR24-GACFAS/weights/resnet18_pICM2O_best.pth")
model.load_state_dict(ckpt["state_dict"])
model.eval()
test_transform = transforms.Compose([
            transforms.Resize((256,256)),
            transforms.ToTensor(),
            transforms.Normalize(mean=[0.485, 0.456, 0.406],
                                         std=[0.229, 0.224, 0.225])
        ])
img = Image.open("/mnt/a/joe/code/STAR/images/video/img_1.png").convert('RGB')
with torch.no_grad():
  input = test_transform(img)
  prob= model(input.unsqueeze(0).cuda())
print(torch.sigmoid(prob))
```

live prob for following 2d screen attack is :0.4750
![image](https://github.com/Leminhbinh0209/CVPR24-FAS/assets/44363764/3d94aa3b-a32b-477a-be4c-b01c53cf8fd4)

however when it comes to real,liveness prob is 0.4721
![image](https://github.com/Leminhbinh0209/CVPR24-FAS/assets/44363764/56f3d591-d170-4fab-a829-ae69ac819b7c)

**Comment by blacksino** [comment]
Thank you for your patience; I'll give it a try. By the way, I'm a newcomer to liveness detection. In my opinion, face cropping might omit some critical information for Facial Anti-Spoofing (FAS) because additional elements visible in the image can assist in determining whether it is spoofed. For instance, bezels could sometimes provide clues.

**Comment by Leminhbinh0209** [comment]
You are right. That could be a good approach if you can utilize it, but it can make a model highly reliant on those background artifacts while training and overfit when evaluated on other different datasets.  In my experience, the most recent models chose to crop the face, but I don't say that your approach is impossible and it just needs more exploration.

**Comment by blacksino** [comment]
Thanks!
